types/is-posix-bracket/index.d.ts

```diff
@@ -0,0 +1,3 @@
+declare function isPosixBracket<T>(str: T): T extends string ? boolean : false;
```

Comment:
It might be worth just typing this as `str: string` instead of having a generic; IMO the original library's `typeof` check is just because it can't provide type hints on its own

```suggestion
declare function isPosixBracket(str: string): boolean;
```